### ISSUE TITLE
Menu.add(T entry, int index) adds to the bottom instead of at index

### ISSUE THREAD

**Issue by hinca** [dorkbox/SystemTray#152]
Using SystemTray 4.1, I have a Swing Menu with 9 entries and am adding another entry to position 4. 

```java
    /**
     * Adds a menu entry, separator, or sub-menu to this menu.
     */
    public
    <T extends Entry> T add(final T entry, final int index) {
        final int insertIndex;
        synchronized (menuEntries) {
            // access on this object must be synchronized for object visibility
            if (index == -1) {
                insertIndex = menuEntries.size();
                menuEntries.add(entry);
            } else {
                if (!menuEntries.isEmpty() && menuEntries.get(0) instanceof Status) {
                    // the "status" menu entry is ALWAYS first
                    insertIndex = index+1;
                } else {
                    insertIndex = index;
                }
                // ************
                menuEntries.add(index, entry); // <---- after this, the entry is in the correct position - Line 236
                // ************
            }
        }

        if (peer != null) {
            // all ADD/REMOVE events have to be queued on our own dispatch thread, so the execution order of the events can be maintained.
            EventDispatch.runLater(()->((MenuPeer) peer).add(Menu.this, entry, insertIndex));
        }

        return entry;
    }
```

This seems to process correctly on the [highlighted line 236](https://github.com/dorkbox/SystemTray/blob/612cceab1f68d7000ad06e121e5f619fb4f54603/src/dorkbox/systemTray/Menu.java#L236), however, when the UI is processed, the entry is in the last position instead of the 4th position.

Could it be because [the index is not used in SwingMenu add implementation](https://github.com/dorkbox/SystemTray/blob/612cceab1f68d7000ad06e121e5f619fb4f54603/src/dorkbox/systemTray/ui/swing/SwingMenu.java#L75)?

**Comment by dorkbox** [comment]
Oh my. I'm not sure, but I'll double check that!

Thanks for the bug report!

**Comment by hinca** [comment]
The current workaround is to remove all the items and add them again on the menu change.

**Comment by ppattard** [comment]
Hello, I can confirm: also ended up with this problem on Windows only (works fine with Gnome for instance) :)

**Comment by dorkbox** [comment]
Thanks for the feedback! I'll have this fixed for the next version.

**Comment by dorkbox** [comment]
Fixed 1a21ad81
